### PR TITLE
Branch 0.4 release docs

### PR DESCRIPTION
**Pull request by sameerz**
1. Update documents for 0.4 release.  Removed -SNAPSHOT or changed plugin 0.3->0.4, cudf 0.17->0.18
1. Updated image version for GCP Dataproc.  Tested 0.4 works with the latest Dataproc Spark 3.1.1 image
1. One broken link fix in docs/additional-functionality/rapids-udfs.md

### CODE REVIEW

File: docs/download.md [hunk 1]
```diff
@@ -4,6 +4,52 @@ title: Download
 nav_order: 3
 ---
 
+## Release v0.4.0
+
+New functionality for the release includes
+* Decimal support up to 64 bit, including reading and writing decimal from Parquet
```
Comment (revans2) [inline docs/download.md:10]: Do we want to say anything about it being disabled by default?

**Review by revans2 — APPROVED** [review]
Just one question

**Review by jlowe — COMMENTED** [review]
Not under the docs directory but documentation all the same, there are references to 0.4.0-SNAPSHOT in integration_tests/README.md and tests/README.md as well.  It's not a requirement to update those in this PR, but they should be updated by the time the pom files are updated.

File: docs/download.md [hunk 2]
```diff
@@ -4,6 +4,52 @@ title: Download
...
+* Ability to call native CUDA or cudf functions from Scala, Java or Hive UDFs
```
Comment (jlowe) [inline docs/download.md:11]:
```suggestion
* Ability for users to provide GPU versions of Scala, Java or Hive UDFs
```

File: docs/download.md [hunk 3]
```diff
@@ -4,6 +4,52 @@ title: Download
...
+* Shuffle and sort support for `struct` data types
+* `array_contains` for list operations
+* `collect_list` and `average` for windowing operations
+* Murmur3 `hash` operation 
+* Reading a v2 datasource when the data being read is an `ArrowColumnVector`
```
Comment (jlowe) [inline docs/download.md:16]:
```suggestion
* Improved performance when reading from DataSource v2 when the source produces data in the Arrow format.
```

File: docs/download.md (hunk 1 above)
Comment (sameerz) [reply docs/download.md]: I added a note on how to enable decimal.  I also updated some text around the decimal config.

File: docs/download.md (hunk 2 above)
Comment (sameerz) [reply docs/download.md]: Incorporated feedback.

File: docs/download.md (hunk 3 above)
Comment (sameerz) [reply docs/download.md]: Incorporated feedback

**Comment by sameerz** [comment]
@jlowe @revans2 @tgravescs I updated the docs, and also added that we support Spark 3.0.2.  

@jlowe I will do a follow up to remove `SNAPSHOT` from integration_tests/README.md and tests/README.md

**Review by revans2 — APPROVED** [review]

**Review by jlowe — APPROVED** [review]

**Comment by sameerz** [comment]
build

**Pull request merged**